pyvo/registry/regtap.py

```diff
@@ -423,16 +424,21 @@ def ivoid2service(ivoid):
         NATURAL JOIN rr.interface
         WHERE ivoid = '{}'
     """.format(tap.escape(ivoid)))
-
+    services = []
     for result in results:
+        id = result["standard_id"].decode()
         for ivo, cls in {
             "ivo://ivoa.net/std/conesearch":  scs.SCSService,
             "ivo://ivoa.net/std/sia":  sia.SIAService,
             "ivo://ivoa.net/std/ssa":  ssa.SSAService,
             "ivo://ivoa.net/std/sla":  sla.SLAService,
             "ivo://ivoa.net/std/tap":  tap.TAPService,
         }.items():
-            if result["standard_id"] in ivo:
-                return cls(result["access_url"])
-
-    return None
+            if id != '' and id in ivo:
+                services.append(cls(result["access_url"].decode()))
+    if len(services) > 1:
+        return services
+    elif len(services) == 1:
+        return services[0]
+    else:
+        return None
```

Comment:
Well, it was in the notebook as an example, so it was once meant to be exposed.  When it broke because it *wasn't* exposed, I put it in the init. 

I completely agree about the oddity of returning either one service or a list.  Thus the plea for better ideas.

Reply:
I've made an update following what msdemlei suggested.